core/src/main/java/tc/oc/pgm/scoreboard/SidebarMatchModule.java

```diff
@@ -470,86 +473,61 @@ private void renderSidebar() {
       }
 
       for (Competitor competitor : sortedCompetitors) {
-        if (!(firstTeam || (isWool() && competitorsWithGoals.size() > 5))) {
+        if (!(firstTeam || isSuperCompactWool)) {
           // Add a blank row between teams
           rows.add("");
         }
         firstTeam = false;
 
-        if (isCompactWool()) {
-          String woolText = " ";
-          boolean firstWool = true;
-
-          List<Goal> sortedWools = new ArrayList<>(gmm.getGoals(competitor));
-          Collections.sort(
-              sortedWools,
-              new Comparator<Goal>() {
-                @Override
-                public int compare(Goal a, Goal b) {
-                  return a.getName().compareToIgnoreCase(b.getName());
-                }
-              });
-
-          boolean inline = false;
-          String altText = " ";
-          int index = 1;
-          for (Goal goal : sortedWools) {
-            if (goal instanceof MonumentWool && goal.isVisible()) {
-              MonumentWool wool = (MonumentWool) goal;
-              if (wool.getDefinition().isInline() == true) {
-                inline = true;
-              }
-              if (!firstWool) {
-                if (sortedWools.size() > 5) {
-                  woolText += " ";
-                  if (index % 2 == 0) {
-                    altText += " ";
+        // Prevent team name from showing up if there is no space for its objectives
+        if (!isSuperCompactWool || rows.size() + 2 < MAX_ROWS) {
+
+          // Add a row for the team name
+          rows.add(competitor.getStyledName(NameStyle.FANCY).render().toLegacyText());
+
+          if (isCompactWool) {
+            String woolText = " ";
+            boolean firstWool = true;
+
+            List<Goal> sortedWools = new ArrayList<>(gmm.getGoals(competitor));
+            Collections.sort(
+                sortedWools,
+                new Comparator<Goal>() {
+                  @Override
+                  public int compare(Goal a, Goal b) {
+                    return a.getName().compareToIgnoreCase(b.getName());
+                  }
+                });
+
+            boolean horizontalCompact =
+                !((MAX_PREFIX + MAX_SUFFIX) < sortedWools.size() + 3 * (sortedWools.size() - 1));
```

Comment:
Leaving a small comment here mentioning why this is done could be helpful for future mantainers